app/services/game-overlay/index.ts

```diff
@@ -0,0 +1,396 @@
+    previewMode: false,
+    opacity: 100,
+    windowProperties: {
+      chat: { position: null, id: null },
+      recentEvents: { position: null, id: null },
+    },
+  };
+
+  windows: {
+    chat: Electron.BrowserWindow;
+    recentEvents: Electron.BrowserWindow;
+    // overlayControls: Electron.BrowserWindow;
+  } = {} as any;
+
+  previewWindows: {
+    chat: Electron.BrowserWindow;
+    recentEvents: Electron.BrowserWindow;
+    // overlayControls: Electron.BrowserWindow;
+  } = {} as any;
+
+  overlayWindow: Electron.BrowserWindow;
+  onWindowsReady: Subject<Electron.BrowserWindow> = new Subject<Electron.BrowserWindow>();
+  onWindowsReadySubscription: Subscription;
+  lifecycle: LoginLifecycle;
+
+  commonWindowOptions = {};
+
+  async initialize() {
+    if (!this.state.isEnabled) return;
+
+    this.lifecycle = await this.userService.withLifecycle({
+      init: this.initializeOverlay,
+      destroy: this.destroyOverlay,
+      context: this,
+    });
+  }
+
+  async initializeOverlay() {
+    overlay.start();
+
+    this.onWindowsReadySubscription = this.onWindowsReady
+      .pipe(
+        take(Object.keys(this.windows).length),
+        delay(5000), // so recent events has time to load
+      )
+      .subscribe({ complete: () => this.createWindowOverlays() });
+
+    this.assignCommonWindowOptions();
+    this.createBrowserWindows();
+    this.createPreviewWindows();
+    await this.configureWindows();
+  }
+
+  assignCommonWindowOptions() {
+    const [containerX, containerY] = this.getWindowContainerStartingPosition();
+    this.commonWindowOptions = {
+      backgroundColor: this.customizationService.themeBackground,
+      show: false,
+      frame: false,
+      width: 300,
+      height: 300,
+      x: containerX,
+      y: containerY,
+      skipTaskbar: true,
+      thickFrame: false,
+      resizable: false,
+      webPreferences: { nodeIntegration: false, contextIsolation: true, offscreen: true },
+    };
+  }
+
+  createBrowserWindows() {
+    this.overlayWindow = new BrowserWindow({
+      ...this.commonWindowOptions,
+      height: 600,
+      width: 600,
+    });
+
+    this.windows.recentEvents = new BrowserWindow({
+      ...this.commonWindowOptions,
+      width: 600,
+      parent: this.overlayWindow,
+    });
+
+    this.windows.chat = new BrowserWindow({
+      ...this.commonWindowOptions,
+      height: 600,
+      parent: this.overlayWindow,
+    });
+
+    // Unneeded window without interactivity
+    // this.windows.overlayControls = this.windowsService.createOneOffWindowForOverlay(
+    //   {
+    //     ...commonWindowOptions,
+    //     // @ts-ignore
+    //     webPreferences: {},
+    //     parent: this.overlayWindow,
+    //     x: containerX - 600,
+    //     y: containerY + 300,
+    //     width: 600,
+    //     height: 300,
+    //     // OneOffWindow options
+    //     isFullScreen: true,
+    //     componentName: 'OverlayWindow',
+    //   },
+    //   'overlay',
+    // );
```

Comment:
IMO let's get rid of all the interactivity stuff that isn't needed in this PR.  Let's use git to track changes we want to add in the future, not comments.